cloud/linode/instances_test.go

```diff
@@ -33,50 +31,36 @@ func TestInstanceExists(t *testing.T) {
 	defer ctrl.Finish()
 
 	client := NewMockClient(ctrl)
-	instances := newInstances(client)
-
-	t.Run("should propagate generic api error", func(t *testing.T) {
-		node := nodeWithProviderID(providerIDPrefix + "123")
-		expectedErr := errors.New("some error")
-		client.EXPECT().GetInstance(gomock.Any(), 123).Times(1).Return(nil, expectedErr)
-
-		exists, err := instances.InstanceExists(ctx, node)
-		assert.ErrorIs(t, err, expectedErr)
-		assert.False(t, exists)
-	})
 
 	t.Run("should return false if linode does not exist (by providerID)", func(t *testing.T) {
+		instances := newInstances(client)
```

Comment:
Had to hoist in the `newInstances` initialisation, otherwise we'd get test pollution due to a shared cache.